test/unit/test-performance.js

```diff
@@ -927,4 +927,214 @@ describes.realWin('PeformanceObserver metrics', {amp: true}, env => {
       env.win.perfMetrics = previousPerfMetrics;
     }
   });
+
+  describe('forwards layout jank metric', () => {
+    it('for browsers that support the visibilitychange event', () => {
+      // Fake the window object so we can control the state
+      // of window.document.visibilityState later on.
+      const fakeWin = {
+        Date: env.win.Date,
+        PerformanceLayoutJank: true,
+        PerformanceObserver: env.sandbox.stub(),
+        addEventListener: env.sandbox.stub(),
+        dispatchEvent: env.win.dispatchEvent,
+        document: {
+          addEventListener: env.sandbox.stub(),
+          hidden: false,
+          readyState: 'complete',
+          removeEventListener: env.sandbox.stub(),
+          // Note: the document starts in a visible state.
+          visibilityState: 'visible',
+        },
+        location: env.win.location,
+        navigator: {
+          // Note: specify an Android Chrome user agent, which supports
+          // the visibilitychange event.
+          userAgent: 'Mozilla/5.0 (Linux; Android 8.0.0; Pixel XL ' +
+          'Build/OPR6.170623.011) AppleWebKit/537.36 (KHTML, like Gecko) ' +
+          'Chrome/61.0.3163.98 Mobile Safari/537.36',
+        },
+        performance: {
+          getEntriesByType: env.sandbox.stub(),
+        },
+        removeEventListener: env.win.removeEventListener,
+      };
```

Comment:
Should be able to simplify some of these by stubbing `isSafari()`. E.g.

```js
let isSafari = false;
sandbox.stub(Services, 'platformFor').returns({isSafari});

// Test non-Safari case.
...

isSafari = true; 

// Test Safari case.
```

Ditto `onVisibilityChanged()` and `isHidden()`.

Reply:
Done in 3dd44e1

Reply:
I should say, done in a fashion I figured out. Does the stubbing make sense?